src/util/taproot.rs

```diff
@@ -646,35 +658,38 @@ impl TaprootMerkleBranch {
         }
     }
 
-    /// Consume Self to get Vec<[`sha256::Hash`]>
+    /// Returns the inner list of hashes.
     pub fn into_inner(self) -> Vec<sha256::Hash> {
         self.0
     }
 }
 
-/// Control Block data structure used in Tapscript satisfaction
+/// Control block data structure used in Tapscript satisfaction.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct ControlBlock {
-    /// The tapleaf version,
+    /// The tapleaf version.
     pub leaf_version: LeafVersion,
-    /// The parity of the output key (NOT THE INTERNAL KEY WHICH IS ALWAYS XONLY)
+    /// The parity of the output key (NOT THE INTERNAL KEY WHICH IS ALWAYS XONLY).
     pub output_key_parity: secp256k1::Parity,
-    /// The internal key
+    /// The internal key.
     pub internal_key: UntweakedPublicKey,
-    /// The merkle proof of a script associated with this leaf
+    /// The merkle proof of a script associated with this leaf.
     pub merkle_branch: TaprootMerkleBranch,
 }
 
 impl ControlBlock {
-    /// Obtain a ControlBlock from slice. This is an extra witness element
-    /// that provides the proof that taproot script pubkey is correctly computed
-    /// with some specified leaf hash. This is the last element in
-    /// taproot witness when spending a output via script path.
+    /// Constructs a `ControlBlock` from slice. This is an extra witness element that provides the
+    /// proof that taproot script pubkey is correctly computed with some specified leaf hash. This
+    /// is the last element in taproot witness when spending a output via script path.
     ///
-    /// # Errors:
-    /// - If the control block size is not of the form 33 + 32m where
-    /// 0 <= m <= 128, InvalidControlBlock is returned
+    /// # Errors
+    ///
+    /// - `TaprootError::InvalidControlBlockSize` if `sl` is not of size 1 + 32 + 32N for any N >= 0.
+    /// - `TaprootError::InvalidParity` if first byte of `sl` is not a valid output key parity.
+    /// - `TaprootError::InvalidTaprootLeafVersion` if first byte of `sl` is not a valid leaf version.
+    /// - `TaprootError::InvalidInternalKey` if internal key is invalid (first 32 bytes after the parity byte).
+    /// - `TaprootErrro::InvalidMerkleTreeDepth` if merkle tree is too deep (more than 128 levels).
```

Comment:
In f488093a04a02573507fcb8bdbd35b4bed7ba7b2:

Type `TaprootErrro`

Reply:
Nice, thanks.